Makefile

```diff
@@ -1,5 +1,7 @@
 .DEFAULT_GOAL := all
 
+LIBDIR=/usr/local/lib
```

Comment:
Can we source in the built-binary to an internal `.build` folder instead?

Reply:
yes, it would be better (and that's how i did originally), but CGo is **very** limited in how to specify paths in the linking variables.  I think it can be done by managing other variables *before* building the Go binding, but not sure how that extends to multiple users of the binding library.   I'm still trying to make it more self-contained in a single script in the binding library that can be easily called by other users.

ah, i don't think we want prebuilt binaries, since that would make targeting multiple plaforms much more complicated.

Reply:
I'm not asking for pre-built binaries here. I'm simply asking to not put the binary in the global path, and instead in a `.build` folder scoped to this project only.

I'm not following why we can't simply change the path here when providing it to CGo... what does the location of the binary have to do with that? I'm simply talking about moving it from `/usr/loca/lib/libatc_router.so` to `/path/to/goks/.build/libatc_router.so`.

Reply:
unfortunately, </path/to/goks> can't be expressed in CGo from another package.  i tried injecting to <local-go-cache/go-atc-router/lib>, but there are complex permission and timing restrictions.  as mentioned, the path might be constructable by manipulating environment variables, but they're hard to propagate from a script to the go build, unless called directly, which is another big change in the build process.

Reply:
Ah, I'm following now. Completely understand, closing out this conversation. Thanks for diving into the details here a bit more.